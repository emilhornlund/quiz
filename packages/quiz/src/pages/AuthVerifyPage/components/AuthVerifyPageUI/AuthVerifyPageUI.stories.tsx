import type { Meta, StoryObj } from '@storybook/react'
import { withRouter } from 'storybook-addon-remix-react-router'

import AuthVerifyPageUI from './AuthVerifyPageUI'

const meta = {
  title: 'Pages/AuthVerifyPage',
  component: AuthVerifyPageUI,
  decorators: [withRouter],
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof AuthVerifyPageUI>

export default meta
type Story = StoryObj<typeof meta>

export const Loading = {
  args: {
    verified: false,
    loggedIn: false,
    error: false,
  },
} satisfies Story

export const Verified = {
  args: {
    verified: true,
    loggedIn: false,
    error: false,
  },
} satisfies Story

export const VerifiedLoggedIn = {
  args: {
    verified: true,
    loggedIn: true,
    error: false,
  },
} satisfies Story

export const Error = {
  args: {
    verified: false,
    loggedIn: false,
    error: true,
  },
} satisfies Story
