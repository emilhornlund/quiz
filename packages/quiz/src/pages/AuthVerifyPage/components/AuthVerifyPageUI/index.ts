export type { AuthVerifyPageUIProps } from './AuthVerifyPageUI'
export { default } from './AuthVerifyPageUI'
