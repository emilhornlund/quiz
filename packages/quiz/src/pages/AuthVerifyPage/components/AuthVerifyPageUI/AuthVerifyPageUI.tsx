import { faCheck, faXmark } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import React, { FC } from 'react'
import { Link } from 'react-router-dom'

import { Badge, LoadingSpinner, Page, Typography } from '../../../../components'

export interface AuthVerifyPageUIProps {
  verified: boolean
  loggedIn: boolean
  error?: boolean
}

const AuthVerifyPageUI: FC<AuthVerifyPageUIProps> = ({
  verified,
  loggedIn,
  error,
}) => {
  return (
    <Page>
      {verified && !error && (
        <>
          <Badge size="large" backgroundColor="green">
            <FontAwesomeIcon icon={faCheck} />
          </Badge>

          <Typography variant="subtitle" size="medium">
            Hooray! Your email’s all set!
          </Typography>

          <Typography variant="text" size="medium">
            Welcome aboard the fun train. Let’s roll!
          </Typography>

          {loggedIn ? (
            <Link to={'/'}>
              <Typography variant="link" size="small">
                Take me home
              </Typography>
            </Link>
          ) : (
            <Link to={'/auth/login'}>
              <Typography variant="link" size="small">
                Log in to get started!
              </Typography>
            </Link>
          )}
        </>
      )}

      {!verified && !error && (
        <>
          <Typography variant="subtitle" size="small">
            One moment… verifying your magic link!
          </Typography>

          <LoadingSpinner />

          <Typography variant="text" size="small">
            Good things come to those who wait!
          </Typography>
        </>
      )}

      {error && (
        <>
          <Badge size="large" backgroundColor="red">
            <FontAwesomeIcon icon={faXmark} />
          </Badge>

          <Typography variant="subtitle" size="small">
            Oops! Something went wrong.
          </Typography>

          <Typography variant="text" size="medium">
            The supplied link is invalid or has expired.
          </Typography>
        </>
      )}
    </Page>
  )
}

export default AuthVerifyPageUI
