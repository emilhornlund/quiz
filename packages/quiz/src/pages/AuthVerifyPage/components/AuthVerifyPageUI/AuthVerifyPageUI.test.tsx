import { render, screen } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import AuthVerifyPageUI from './AuthVerifyPageUI'

describe('AuthVerifyPageUI', () => {
  it('renders verifying state', () => {
    const { container } = render(
      <MemoryRouter>
        <AuthVerifyPageUI verified={false} loggedIn={false} />
      </MemoryRouter>,
    )

    expect(
      screen.getByText('One moment… verifying your magic link!'),
    ).toBeInTheDocument()
    expect(
      screen.getByText('Good things come to those who wait!'),
    ).toBeInTheDocument()
    expect(screen.queryByText('Hooray! Your email’s all set!')).toBeNull()
    expect(screen.queryByText('Oops! Something went wrong.')).toBeNull()

    expect(container).toMatchSnapshot()
  })

  it('renders verified state (not logged in) with login link', () => {
    const { container } = render(
      <MemoryRouter>
        <AuthVerifyPageUI verified={true} loggedIn={false} />
      </MemoryRouter>,
    )

    expect(
      screen.getByText('Hooray! Your email’s all set!'),
    ).toBeInTheDocument()
    expect(
      screen.getByText('Welcome aboard the fun train. Let’s roll!'),
    ).toBeInTheDocument()

    const loginLink = container.querySelector('a[href="/auth/login"]')
    expect(loginLink).toBeTruthy()
    expect(screen.getByText('Log in to get started!')).toBeInTheDocument()

    // no home link in this branch
    expect(container.querySelector('a[href="/"]')).toBeNull()

    expect(container).toMatchSnapshot()
  })

  it('renders verified state (logged in) with home link', () => {
    const { container } = render(
      <MemoryRouter>
        <AuthVerifyPageUI verified={true} loggedIn={true} />
      </MemoryRouter>,
    )

    expect(
      screen.getByText('Hooray! Your email’s all set!'),
    ).toBeInTheDocument()
    expect(
      screen.getByText('Welcome aboard the fun train. Let’s roll!'),
    ).toBeInTheDocument()

    const home = screen.getByRole('link', { name: 'Take me home' })
    expect(home).toHaveAttribute('href', '/')

    // ensure the *component’s* login link isn’t present
    expect(
      screen.queryByRole('link', { name: 'Log in to get started!' }),
    ).toBeNull()

    expect(container).toMatchSnapshot()
  })

  it('renders error state', () => {
    const { container } = render(
      <MemoryRouter>
        <AuthVerifyPageUI verified={false} loggedIn={false} error={true} />
      </MemoryRouter>,
    )

    expect(screen.getByText('Oops! Something went wrong.')).toBeInTheDocument()
    expect(
      screen.getByText('The supplied link is invalid or has expired.'),
    ).toBeInTheDocument()

    // no verifying or success texts in error branch
    expect(
      screen.queryByText('One moment… verifying your magic link!'),
    ).toBeNull()
    expect(screen.queryByText('Hooray! Your email’s all set!')).toBeNull()

    expect(container).toMatchSnapshot()
  })
})
