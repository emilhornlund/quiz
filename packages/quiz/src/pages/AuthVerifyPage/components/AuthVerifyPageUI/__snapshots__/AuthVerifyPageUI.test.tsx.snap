// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthVerifyPageUI > should render AuthVerifyPageUI as loading 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="content fullWidth centerAlign"
    >
      <div
        class="typography subtitle small"
      >
        One moment… verifying your magic link!
      </div>
      <div
        class="loadingSpinnerContainer"
      >
        <div />
        <div />
        <div />
      </div>
      <div
        class="typography text small"
      >
        Good things come to those who wait!
      </div>
    </div>
  </div>
</div>
`;

exports[`AuthVerifyPageUI > should render AuthVerifyPageUI as verified 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="content fullWidth centerAlign"
    >
      <div
        class="badge large green"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check "
          data-icon="check"
          data-prefix="fas"
          role="img"
          viewBox="0 0 448 512"
        >
          <path
            d="M434.8 70.1c14.3 10.4 17.5 30.4 7.1 44.7l-256 352c-5.5 7.6-14 12.3-23.4 13.1s-18.5-2.7-25.1-9.3l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l101.5 101.5 234-321.7c10.4-14.3 30.4-17.5 44.7-7.1z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="typography subtitle medium"
      >
        Hooray! Your email’s all set!
      </div>
      <div
        class="typography text medium"
      >
        Welcome aboard the fun train. Let’s roll!
      </div>
      <a
        data-discover="true"
        href="/auth/login"
      >
        <div
          class="typography link small"
        >
          Log in to get started!
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`AuthVerifyPageUI > should render AuthVerifyPageUI as verified and logged in 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="content fullWidth centerAlign"
    >
      <div
        class="badge large green"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check "
          data-icon="check"
          data-prefix="fas"
          role="img"
          viewBox="0 0 448 512"
        >
          <path
            d="M434.8 70.1c14.3 10.4 17.5 30.4 7.1 44.7l-256 352c-5.5 7.6-14 12.3-23.4 13.1s-18.5-2.7-25.1-9.3l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l101.5 101.5 234-321.7c10.4-14.3 30.4-17.5 44.7-7.1z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="typography subtitle medium"
      >
        Hooray! Your email’s all set!
      </div>
      <div
        class="typography text medium"
      >
        Welcome aboard the fun train. Let’s roll!
      </div>
      <a
        data-discover="true"
        href="/"
      >
        <div
          class="typography link small"
        >
          Take me home
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`AuthVerifyPageUI > should render AuthVerifyPageUI with error 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="content fullWidth centerAlign"
    >
      <div
        class="badge large red"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark "
          data-icon="xmark"
          data-prefix="fas"
          role="img"
          viewBox="0 0 384 512"
        >
          <path
            d="M55.1 73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L147.2 256 9.9 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192.5 301.3 329.9 438.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.8 256 375.1 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192.5 210.7 55.1 73.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="typography subtitle small"
      >
        Oops! Something went wrong.
      </div>
      <div
        class="typography text medium"
      >
        The supplied link is invalid or has expired.
      </div>
    </div>
  </div>
</div>
`;
